includes/class-duplicate-payment-prevention-service.php

```diff
@@ -102,9 +102,13 @@ public function check_payment_intent_attached_to_order_succeeded( WC_Order $orde
 			return;
 		}
 
-		$intent_meta_order_id_raw = $intent->get_metadata()['order_id'] ?? '';
-		$intent_meta_order_id     = is_numeric( $intent_meta_order_id_raw ) ? intval( $intent_meta_order_id_raw ) : 0;
-		if ( $intent_meta_order_id !== $order->get_id() ) {
+		$intent_meta_order_id_raw     = $intent->get_metadata()['order_id'] ?? '';
+		$intent_meta_order_id         = is_numeric( $intent_meta_order_id_raw ) ? intval( $intent_meta_order_id_raw ) : 0;
+		$intent_meta_order_number_raw = $intent->get_metadata()['order_number'] ?? '';
+		$intent_meta_order_number     = is_numeric( $intent_meta_order_number_raw ) ? intval( $intent_meta_order_number_raw ) : 0;
+		$paid_on_woopay               = filter_var( $intent->get_metadata()['paid_on_woopay'] ?? false, FILTER_VALIDATE_BOOLEAN );
+		$is_woopay_order              = $order->get_id() === $intent_meta_order_number;
+		if ( ! ( $paid_on_woopay && $is_woopay_order ) && $intent_meta_order_id !== $order->get_id() ) {
```

Comment:
Nit: No changes are necessary but what do you think about renaming the variables as follows:

```suggestion
		$intent_meta_order_id_raw      = $intent->get_metadata()['order_id'] ?? '';
		$intent_meta_merchant_order_id = is_numeric( $intent_meta_order_id_raw ) ? intval( $intent_meta_order_id_raw ) : 0;
		$intent_meta_order_number_raw  = $intent->get_metadata()['order_number'] ?? '';
		$intent_meta_woopay_order_id   = is_numeric( $intent_meta_order_number_raw ) ? intval( $intent_meta_order_number_raw ) : 0;
		$paid_on_woopay                = filter_var( $intent->get_metadata()['paid_on_woopay'] ?? false, FILTER_VALIDATE_BOOLEAN );
		$is_woopay_order               = $order->get_id() === $intent_meta_woopay_order_id;
		$is_merchant_order             = $intent_meta_merchant_order_id === $order->get_id();
		if ( ! ( $paid_on_woopay && $is_woopay_order ) && ! $is_merchant_order ) {
```

Reply:
I don't think this renaming is beneficial because depending on the context we're analyzing the intent order ID may belong to WooPay order instead of the merchant's, and that may cause confusion. I only named `is_woopay_order` like that because it seems the `order_number` was created especially to distinguish WooPay orders and merchant orders.